app/components/SelectWallet.tsx

```diff
@@ -1,81 +1,132 @@
 import { useContext } from "react";
 import { useLocation } from "react-router";
-import { BitcoinIcon, Wallet } from "lucide-react";
-import { Dialog, DialogContent, DialogTrigger } from "~/components/ui/dialog";
-import { XverseWallet } from "./Wallet/XverseWallet/XverseWallet";
-import { SuiWallet } from "./Wallet/SuiWallet/SuiWallet";
-import { SuiModal } from "./Wallet/SuiWallet/SuiModal";
+import { BitcoinIcon, Wallet, Bitcoin } from "lucide-react";
+import { SuiConnectModal } from "./Wallet/SuiWallet/SuiModal";
 import { WalletContext } from "~/providers/ByieldWalletProvider";
 import { Wallets } from "~/components/Wallet";
 import { useXverseConnect } from "./Wallet/XverseWallet/useWallet";
 import { routes } from "~/config/walletVisibility";
-
-function WalletSection({ title, children }: { title: string; children: React.ReactNode }) {
-	return (
-		<div className="space-y-2">
-			<h4 className="text-sm font-medium">{title}</h4>
-			{children}
-		</div>
-	);
-}
-
-function MobileWalletModal({ children }: { children: React.ReactNode }) {
-	return (
-		<div className="md:hidden">
-			<Dialog>
-				<DialogTrigger asChild>
-					<button className="btn btn-primary btn-outline btn-sm">
-						<Wallet size={16} /> Wallet
-					</button>
-				</DialogTrigger>
-				<DialogContent className="sm:max-w-md">
-					<div className="space-y-4">{children}</div>
-				</DialogContent>
-			</Dialog>
-		</div>
-	);
-}
-
-function DesktopWallets({ children }: { children: React.ReactNode }) {
-	return <div className="hidden md:flex md:items-center md:gap-4">{children}</div>;
-}
+import { trimAddress } from "~/components/Wallet/walletHelper";
+import { useXverseWallet } from "~/components/Wallet/XverseWallet/useWallet";
+import { SUIIcon } from "~/components/icons";
+import { useCurrentAccount, useAccounts, useSwitchAccount } from "@mysten/dapp-kit";
 
 export function SelectWallet() {
 	const { isWalletConnected } = useContext(WalletContext);
 	const { connectWallet } = useXverseConnect();
 	const { pathname } = useLocation();
 
+	const { currentAddress, setCurrentAddress, addressInfo: bitcoinAddresses } = useXverseWallet();
+	const currentSuiAccount = useCurrentAccount();
+	const allSuiAccounts = useAccounts();
+	const { mutate: switchSuiAccount } = useSwitchAccount();
+
 	const shouldShowBitcoinWallet = routes[pathname]?.bitcoin ?? true;
 	const shouldShowSUIWallet = routes[pathname]?.sui ?? true;
+
 	const isBitcoinConnected = isWalletConnected(Wallets.Xverse);
 	const isSuiConnected = isWalletConnected(Wallets.SuiWallet);
 
-	const bitcoinWallet =
+	const bitcoinLoading = isBitcoinConnected && !currentAddress;
+	const suiLoading = isSuiConnected && !currentSuiAccount;
+
+	// If still loading for connected wallets, show loading indicators
+	// TODO: this have to be finished
```

Comment:
The TODO I left here is to update the logic:
- when page / wallet is loading , we should not show "Connect Sui" (nor Bitcoin), instead we the wallet badge should show "Loading wallet"
- when wallet status is loaded then if wallet is not connected, we should show the connect wallet button. Otherwise It should be a switch account.

Reply:
Done